cps/main.c

```diff
@@ -1088,9 +1092,24 @@ run_cps(struct net_config *net_conf, struct gk_config *gk_conf,
 		goto stage3;
 	}
 
+	if (gk_conf != NULL) {
+		cps_conf->mailbox_max_pkt_burst =
+			RTE_MAX(gk_conf->front_max_pkt_burst,
+				gk_conf->back_max_pkt_burst);
+	}
+
+	if (gt_conf != NULL) {
+		cps_conf->mailbox_max_pkt_burst =
+			RTE_MAX(cps_conf->mailbox_max_pkt_burst,
+				gt_conf->gt_max_pkt_burst);
+	}
+
+	ele_size = RTE_MAX(sizeof(struct cps_request),
+		sizeof(enum cps_req_ty) + sizeof(struct cps_bgp_req) +
+		sizeof(struct rte_mbuf *) * cps_conf->mailbox_max_pkt_burst);
```

Comment:
Using `sizeof(enum cps_req_ty)` is fragile. If a field is added to `struct cps_request`, things can silently break. This would be a hard bug to catch.

My suggestion is as follows:
1. Add the following field just before the `union` of `struct cps_request`:
`int end_of_header[0];`
2. Use `offsetof(struct cps_request, end_of_header)` instead of `sizeof(enum cps_req_ty)`.